src/providers/ProviderFactory.ts

```diff
@@ -3,30 +3,31 @@ import { AwsStorage } from '../providers/aws/AwsStorage';
 import { GoogleStorage } from '../providers/google/GoogleStorage';
 import { AzureStorage } from '../providers/azure/AzureStorage';
 import { IProviderConfig } from '../interfaces/IProviderConfig';
+import { ErrorHandler } from '../utils/ErrorHandler';
 
 export class ProviderFactory {
 	static createProvider (config: IProviderConfig): IStorageProvider {
 		switch (config.provider) {
 			case 'aws':
 				if (!config.aws) {
-					throw new Error('AWS configuration is required');
+					throw new ErrorHandler('AwsProvider', 'AWS configuration is required', 401)
```

Comment:
All of these changes not needed here; these should throw regular errors.
Only the errors from the provider requests should be handled exceptionally with the ErrorHandler class